integration-tests/artifacts/__tests__/static-queries.js

```diff
@@ -0,0 +1,182 @@
+  })
+
+  test(`are written correctly in tsx`, async () => {
+    const queries = [titleQuery, ...globalQueries]
+    const pagePath = `/tsx/`
+
+    const { staticQueryHashes } = await readPageData(publicDir, pagePath)
+
+    expect(staticQueryHashes.sort()).toEqual(queries.map(hashQuery).sort())
+  })
+
+  test(`are written correctly in typescript`, async () => {
+    const queries = [titleQuery, ...globalQueries]
+    const pagePath = `/typescript/`
+
+    const { staticQueryHashes } = await readPageData(publicDir, pagePath)
+
+    expect(staticQueryHashes.sort()).toEqual(queries.map(hashQuery).sort())
+  })
+
+  test(`are written correctly when nesting imports`, async () => {
+    const queries = [titleQuery, authorQuery, ...globalQueries]
+    const pagePath = `/import-import/`
+
+    const { staticQueryHashes } = await readPageData(publicDir, pagePath)
+
+    expect(staticQueryHashes.sort()).toEqual(queries.map(hashQuery).sort())
+  })
+
+  test(`are written correctly when nesting dynamic imports`, async () => {
+    const queries = [titleQuery, ...globalQueries]
+    const pagePath = `/dynamic-dynamic/`
+
+    const { staticQueryHashes } = await readPageData(publicDir, pagePath)
+
+    expect(staticQueryHashes.sort()).toEqual(queries.map(hashQuery).sort())
+  })
+
+  test(`are written correctly when nesting a dynamic import in a regular import`, async () => {
+    const queries = [titleQuery, authorQuery, ...globalQueries]
+    const pagePath = `/import-dynamic/`
+
+    const { staticQueryHashes } = await readPageData(publicDir, pagePath)
+
+    expect(staticQueryHashes.sort()).toEqual(queries.map(hashQuery).sort())
+  })
+
+  test(`are written correctly when nesting a regular import in a dynamic import`, async () => {
+    const queries = [titleQuery, ...globalQueries]
+    const pagePath = `/dynamic-import/`
+
+    const { staticQueryHashes } = await readPageData(publicDir, pagePath)
+
+    expect(staticQueryHashes.sort()).toEqual(queries.map(hashQuery).sort())
+  })
+
+  test("are written correctly with circular dependency", async () => {
+    const queries = [titleQuery, ...globalQueries]
+    const pagePath = `/circular-dep/`
+
+    const { staticQueryHashes } = await readPageData(publicDir, pagePath)
+
+    expect(staticQueryHashes.sort()).toEqual(queries.map(hashQuery).sort())
+  })
+
+  // test(`are written correctly when using wrapRootElement`, async () => {
+  //   const queries = [titleQuery]
+  //   const pagePath = `/dynamic-import/`
+
+  //   const { staticQueryHashes } = await readPageData(publicDir, pagePath)
+
+  //   expect(staticQueryHashes.sort()).toEqual(queries.map(hashQuery).sort())
+  // })
+
+  // test(`are written correctly when using wrapPageElement`, async () => {
+  //   const queries = [titleQuery]
+  //   const pagePath = `/dynamic-import/`
+
+  //   const { staticQueryHashes } = await readPageData(publicDir, pagePath)
+
+  //   expect(staticQueryHashes.sort()).toEqual(queries.map(hashQuery).sort())
+  // })
```

Comment:
shouldn't we enable these?

Reply:
Could you tell me why we don't enable them?

Reply:
I've enabled them, or like one of them cause they are redundant.

Reply:
Why not remove them?

Reply:
We should remove them. They are indeed redundant.